### PR TITLE
package/sdbus-c++: add package at v0.8.2

### PR DESCRIPTION
**Pull request by dleeds-cpi**
Adds Buildroot support for this package: https://github.com/Kistler-Group/sdbus-cpp

I haven't verified the host tools build for this. Does it look okay?

### CODE REVIEW

File: package/sdbus-cpp/Config.in [hunk 1]
```diff
@@ -0,0 +1,18 @@
+config BR2_PACKAGE_SDBUS_CPP
+	bool "sdbus-c++"
+	depends on BR2_INSTALL_LIBSTDCPP
+	depends on BR2_PACKAGE_SYSTEMD
```
Comment (vgahlan-cpi) [inline package/sdbus-cpp/Config.in:4]: You may need googletest dependency if you want to enable standalone test.

File: package/sdbus-cpp/Config.in.host [hunk 2]
```diff
@@ -0,0 +1,11 @@
+config BR2_PACKAGE_HOST_SDBUS_CPP
+	bool "sdbus-c++-xml2cpp"
+	select BR2_PACKAGE_HOST_PKGCONF
```
Comment (chardin-cpi) [inline package/sdbus-cpp/Config.in.host:3]: you don't need this - just add in the "host-pkgconf" to the Dependencies - it get's build from there. You do selects when you want those tools but nothing depends on them for like a script or something like that.

File: package/sdbus-cpp/sdbus-cpp.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
+################################################################################
+#
+# SDBUS_CPP
```
Comment (chardin-cpi) [inline package/sdbus-cpp/sdbus-cpp.mk:3]: lower case - that is the boiler template - sdbuss-cpp

File: package/sdbus-cpp/sdbus-cpp.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#
+################################################################################
+
+SDBUS_CPP_VERSION = 0.8.2
+SDBUS_CPP_SITE = $(call github,Kistler-Group,sdbus-cpp,v$(SDBUS_CPP_VERSION))
+SDBUS_CPP_LICENSE = LGPL-2.1
+SDBUS_CPP_LICENSE_FILES = COPYING
+SDBUS_CPP_DEPENDENCIES = host-pkgconf systemd
```
Comment (chardin-cpi) [inline package/sdbus-cpp/sdbus-cpp.mk:11]: why is this not dependent on the host-sdbuss-cpp?

File: package/sdbus-cpp/sdbus-cpp.mk [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+SDBUS_CPP_INSTALL_STAGING = YES
+	
+HOST_SDBUS_CPP_DEPENDENCIES = host-pkgconf
+HOST_SDBUS_CPP_CONF_OPTS += \
+	-DBUILD_CODE_GEN=ON \
+	-DBUILD_DOC=OFF \
+	-DBUILD_TESTS=OFF \
+	-DBUILD_LIBSYSTEMD=ON
+
+$(eval $(cmake-package))
```
Comment (chardin-cpi) [inline package/sdbus-cpp/sdbus-cpp.mk:26]: $(eval $(host-cmake-package))

You need that if you are building the host tool

**Comment by chardin-cpi** [comment]
> Looks good to me. Also remove the other sdbusplus package to avoid any confusion.

Please do not remove the other package - that is upstreamed in buildroot already and will just get winked back in on merges from upstream. Just make sure this is a different name and it is a different package already - if you want to post the patch upstream you can ask if they want to obsolete one vs the other, but likely won't for the time being.

File: package/sdbus-cpp/sdbus-cpp.mk (hunk 4 above)
Comment (dleeds-cpi) [reply package/sdbus-cpp/sdbus-cpp.mk:11]: The host tool is a code generator. If we use it in our emb-core build, we'll need it available.

File: package/sdbus-cpp/sdbus-cpp.mk (hunk 5 above)
Comment (dleeds-cpi) [reply package/sdbus-cpp/sdbus-cpp.mk:26]: Thanks!

**Comment by dleeds-cpi** [comment]
@chardin-cpi do I need `HOST_SDBUS_CPP_INSTALL_STAGING = YES`? to get the binary into the host sysroot /usr/bin?

**Comment by vgahlan-cpi** [comment]
> Please do not remove the other package - that is upstreamed in buildroot already and will just get winked back in on merges from upstream. Just make sure this is a different name and it is a different package already - if you want to post the patch upstream you can ask if they want to obsolete one vs the other, but likely won't for the time being.

Ok, If that is upstream package and not being enabled in configuration. I thought that was added locally and being part of release.

File: package/sdbus-cpp/Config.in (hunk 1 above)
Comment (dleeds-cpi) [reply package/sdbus-cpp/Config.in:4]: The test suite is disabled in the buildroot package.

**Comment by dleeds-cpi** [comment]
Any further changes requested?

**Comment by chardin-cpi** [comment]
> @chardin-cpi do I need `HOST_SDBUS_CPP_INSTALL_STAGING = YES`? to get the binary into the host sysroot /usr/bin?

No - host tools only go into "host" - the staging and target are the cross compiled tools

File: package/sdbus-cpp/Config.in (hunk 1 above)
Comment (chardin-cpi) [reply package/sdbus-cpp/Config.in:4]: that's expected and fine

**Comment by dleeds-cpi** [comment]
@chardin-cpi after you directed me to enable the host build, now it's breaking. Looks like the external libsystemd build is broken in this context. I don't see what it's missing, just yet.

```
>>> host-sdbus-cpp 0.8.2 Building
PATH="/home/dleeds/workspace/emb-platform/BUILD/panther/ucb/host/bin:/home/dleeds/workspace/emb-platform/BUILD/panther/ucb/host/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" PKG_CONFIG="/home/dleeds/workspace/emb-platform/BUILD/panther/ucb/host/bin/pkg-config" PKG_CONFIG_SYSROOT_DIR="/" PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 PKG_CONFIG_ALLOW_SYSTEM_LIBS=1 PKG_CONFIG_LIBDIR="/home/dleeds/workspace/emb-platform/BUILD/panther/ucb/host/lib/pkgconfig:/home/dleeds/workspace/emb-platform/BUILD/panther/ucb/host/share/pkgconfig"  /usr/bin/make -j9  -C /home/dleeds/workspace/emb-platform/BUILD/panther/ucb/build/host-sdbus-cpp-0.8.2/
Scanning dependencies of target sdbus-c++-xml2cpp
Scanning dependencies of target LibsystemdBuildProject
[  4%] Creating directories for 'LibsystemdBuildProject'
[  8%] Building CXX object tools/CMakeFiles/sdbus-c++-xml2cpp.dir/xml2cpp-codegen/xml2cpp.cpp.o
[ 12%] Building CXX object tools/CMakeFiles/sdbus-c++-xml2cpp.dir/xml2cpp-codegen/xml.cpp.o
[ 16%] Building CXX object tools/CMakeFiles/sdbus-c++-xml2cpp.dir/xml2cpp-codegen/generator_utils.cpp.o
[ 20%] Building CXX object tools/CMakeFiles/sdbus-c++-xml2cpp.dir/xml2cpp-codegen/AdaptorGenerator.cpp.o
[ 28%] Building CXX object tools/CMakeFiles/sdbus-c++-xml2cpp.dir/xml2cpp-codegen/BaseGenerator.cpp.o
[ 28%] Building CXX object tools/CMakeFiles/sdbus-c++-xml2cpp.dir/xml2cpp-codegen/ProxyGenerator.cpp.o
[ 32%] Performing download step (git clone) for 'LibsystemdBuildProject'
[ 36%] Linking CXX executable sdbus-c++-xml2cpp
[ 36%] Built target sdbus-c++-xml2cpp
-- LibsystemdBuildProject download command succeeded.  See also /home/dleeds/workspace/emb-platform/BUILD/panther/ucb/build/host-sdbus-cpp-0.8.2/libsystemd-v242/src/LibsystemdBuildProject-stamp/LibsystemdBuildProject-download-*.log
[ 44%] No update step for 'LibsystemdBuildProject'
[ 44%] No patch step for 'LibsystemdBuildProject'
[ 48%] Performing configure step for 'LibsystemdBuildProject'
CMake Error at /home/dleeds/workspace/emb-platform/BUILD/panther/ucb/build/host-sdbus-cpp-0.8.2/libsystemd-v242/src/LibsystemdBuildProject-stamp/LibsystemdBuildProject-configure-.cmake:49 (message):
  Command failed: 1

   '/home/dleeds/workspace/emb-platform/BUILD/panther/ucb/host/bin/cmake' '-Dmake=' '-Dconfig=' '-P' '/home/dleeds/workspace/emb-platform/BUILD/panther/ucb/build/host-sdbus-cpp-0.8.2/libsystemd-v242/src/LibsystemdBuildProject-stamp/LibsystemdBuildProject-configure--impl.cmake'

  See also

    /home/dleeds/workspace/emb-platform/BUILD/panther/ucb/build/host-sdbus-cpp-0.8.2/libsystemd-v242/src/LibsystemdBuildProject-stamp/LibsystemdBuildProject-configure-*.log


CMakeFiles/LibsystemdBuildProject.dir/build.make:107: recipe for target 'libsystemd-v242/src/LibsystemdBuildProject-stamp/LibsystemdBuildProject-configure' failed
```

**Pull request merged**